enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -919,11 +919,25 @@ func (s *ExecutionServer) updateUsage(ctx context.Context, cmd *repb.Command, ex
 		return err
 	}
 	counts := &tables.UsageCounts{}
+	// TODO: Incorporate remote-header overrides here.
 	plat := platform.ParseProperties(&repb.ExecutionTask{Command: cmd})
+
+	executorGroupID, _, err := s.env.GetSchedulerService().GetGroupIDAndDefaultPoolForUser(ctx, plat.OS, plat.UseSelfHostedExecutors)
```

Comment:
Maybe a slightly cleaner approach would be to have the scheduler service return a struct with this information as well as indicate whether we are using shared executors.

Reply:
Done, PTAL. Not super thrilled with the naming, lmk if anything comes to mind that seems more appropriate.

Reply:
Only other alternative I can think of is `PoolMetadata`/`GetPoolMetadata` but not sure that's much better.